os/board/common/Kconfig

```diff
@@ -11,6 +11,19 @@ config ARCH_BOARD_HAVE_FLASH
 		This is enabled by board config.
 
 if ARCH_BOARD_HAVE_FLASH && (NFILE_DESCRIPTORS != 0)
+
+config FLASH_START_ADDR
```

Comment:
1. They should not be in `(NFILE_DESCRIPTORS != 0)`. Let's take them out.
2. If config is invisible, it is set as default value when menuconfig is executed. Then, when we run menuconfig, we should modify this value every time. So, you should put them in board-specific Kconfig.